src/openms/include/OpenMS/ANALYSIS/ID/FIAMSDataProcessor.h

```diff
@@ -0,0 +1,209 @@
+#include <OpenMS/FILTERING/NOISEESTIMATION/SignalToNoiseEstimatorMedianRapid.h>
+#include <OpenMS/FILTERING/BASELINE/MorphologicalFilter.h>
+#include <OpenMS/KERNEL/SpectrumHelper.h>
+#include <OpenMS/ANALYSIS/OPENSWATH/SpectrumAddition.h>
+#include <OpenMS/FILTERING/SMOOTHING/SavitzkyGolayFilter.h>
+#include <OpenMS/FORMAT/MzMLFile.h>
+
+using namespace std;
+
+namespace OpenMS
+{
+  /**
+    description
+  */
+  class OPENMS_DLLAPI FIAMSDataProcessor 
+  {
+public:
+    /// Default constructor
+    FIAMSDataProcessor(
+      String filename, 
+      String dir_input, 
+      String dir_output, 
+      float resolution, 
+      String polarity, 
+      String db_mapping, 
+      String db_struct, 
+      String positive_adducts, 
+      String negative_adducts, 
+      bool store_progress=true,
+      float min_mz=50, 
+      float max_mz=1500, 
+      float bin_step=20
+    );
+
+    /// Default desctructor
+    ~FIAMSDataProcessor();
+
+    /// Copy constructor
+    FIAMSDataProcessor(const FIAMSDataProcessor& cp);
+
+    /// Assignment
+    FIAMSDataProcessor& operator=(const FIAMSDataProcessor& fdp);
+
+    /// Process the given file
+    void run(float n_seconds, OpenMS::MzTab & output);
+
+    /// Cut the spectra for time
+    void cutForTime(const MSExperiment & experiment, vector<MSSpectrum> & output, float n_seconds=6000);
+
+    /// Merge spectra from different retention times into one
+    MSSpectrum mergeAlongTime(const std::vector<OpenMS::MSSpectrum> & input);
+
+    /// Pick peaks from merged spectrum and return as featureMap with the corresponding polarity
+    MSSpectrum extractPeaks(const MSSpectrum & input);
+
+    /// Convert a spectrum to a feature map with the corresponding polarity
+    FeatureMap convertToFeatureMap(const MSSpectrum & input);
+
+    /// Estimate noise for each peak
+    MSSpectrum trackNoise(const MSSpectrum & input);
+
+    /// Perform accurate mass search
+    void runAccurateMassSearch(FeatureMap & input, OpenMS::MzTab & output);
+
+    /// Get filename
+    const String getFilename();
+
+    /// Get input directory
+    const String getInputDir();
+
+    /// Get output directory
+    const String getOutputDir();
+
+    /// Get resolution
+    const float getResolution();
+
+    /// Get resolution
+    const String getPolarity();
+
+    /// Get minimum mass-to-charge
+    const float getMinMZ();
+
+    /// Get maximum mass-to-charge
+    const float getMaxMZ();
+
+    /// Get the sliding bin step
+    const float getBinStep();
+
+    /// Get the path to the db:mapping for passing to AccurateMassSearch
+    const String getDBMapping();
+
+    /// Get the path to the db:struct for passing to AccurateMassSearch
+    const String getDBStruct();
+
+    /// Get the path to the positive adducts for passing to AccurateMassSearch
+    const String getPositiveAdducts();
+
+    /// Get the path to the negative adducts for passing to AccurateMassSearch
+    const String getNegativeAdducts();
```

Comment:
have you thought about using the `DefaultParamHandler` instead of using these class members and all the `get` methods?